libtock/tock.c

```diff
@@ -277,19 +322,43 @@ void yield(void) {
     task_cur = (task_cur + 1) % TASK_QUEUE_SIZE;
     task.cb(task.arg0, task.arg1, task.arg2, task.ud);
   } else {
+    register uint32_t a0  asm ("a0") = 1; // yield-wait
     asm volatile (
-      "li    a4, 0\n"
+      "li    a5, 0\n"
       "ecall\n"
       :
-      :
-      : "memory", "a0", "a1", "a2", "a3", "a4", "a5", "a6", "a7",
+      : "r" (a0)
```

Comment:
the ARM implementation of yield passes both r0 and r1 as inputs, but the risc-v one (this one) does not. I am not sure if it matters, since yield should not use the r1 argument, but it would be nice to use the same approach for ARM and risc-v

Reply:
Agreed.